### PR TITLE
Ensuring service resolution fallback factory is properly configured to avoid disposal

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: src/WebJobs.Script.WebHost/DependencyInjection/ScopedResolver.cs [hunk 1]
```diff
@@ -32,7 +34,7 @@ public void Dispose()
                 .ContinueWith(t =>
                 {
                     Container.Dispose();
-                });
+                }, TaskContinuationOptions.ExecuteSynchronously);
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/DependencyInjection/ScopedResolver.cs]: This addresses potential issues with disposal timing. Added a test to highlight the problem and ensure we don't regress.

Comment (brettsam) [reply src/WebJobs.Script.WebHost/DependencyInjection/ScopedResolver.cs]: Which test covers this?

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/DependencyInjection/ScopedResolver.cs]: Discussed offline.

File: test/WebJobs.Script.Tests/DependencyInjection/JobHostServiceProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            TestService rootService = scope.ServiceProvider.GetService<TestService>();
+
+            scope.Dispose();
+
+            // Disposing of the JobHost scope should also trigger a parent scope disposal
+            Assert.True(rootService.Disposed);
+        }
+
+        [Fact]
+        public void Dispose_OnJobHost_DoesNotDisposRootScopedService()
```
Comment (ankitkumarr) [inline test/WebJobs.Script.Tests/DependencyInjection/JobHostServiceProviderTests.cs]: super nit: "Dispose_OnJobHost_DoesNotDisposRootScopedService" -> "Dispose_OnJobHost_DoesNotDispos**e**RootScopedService"

Comment (fabiocav) [reply test/WebJobs.Script.Tests/DependencyInjection/JobHostServiceProviderTests.cs]: Good catch... just saw this after the merge. Will fix

**Pull request merged**